site/plugins/cdn/src/Image.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+namespace Kirby\Cdn;
+
+use Kirby\Cms\App;
+use Kirby\Cms\File;
+use Kirby\Http\Url;
+use Kirby\Image\Darkroom;
+use Kirby\Image\Focus;
+
+/**
+ * Helper class for KeyCDN image processing
+ *
+ * @package   Kirby Cdn
+ * @author    Lukas Bestle <lukas@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://opensource.org/licenses/MIT
+ */
+class Image
+{
+	/**
+	 * Generates a URL with KeyCDN image processing parameters
+	 * (https://www.keycdn.com/support/image-processing)
+	 * based on a file URL or object and Kirby thumb params
+	 */
+	public static function url(string|File $file, array $params = []): string
+	{
+		$kirby = App::instance();
+
+		$root = $file;
+		if (is_object($file) === true) {
+			$root = $file->root();
+			$file = $file->mediaUrl();
+		}
+
+		$path = Url::path($file);
+
+		$darkroom = Darkroom::factory(
+			'im',
+			$kirby->option('thumbs', [])
+		);
+		$params = $darkroom->preprocess($root, $params);
```

Comment:
`Suggestion:` Move lines 39-43 to its own (maybe protected) method, e.g. `->options()` or `->params()`.

Reply:
Hm, then we need another `$kirby = App::instance()` line. Also I think it adds too much boilerplate code without adding a benefit on its own. What are your thoughts behind the suggestion?

Reply:
Mainly making the `url()` method even easier to read. For understanding its main task, I'd only need to know that these are the options, not how the options are gathered via the Darkroom class. If I'd want to know that, I could look into the method that handles gathering/preprocessing the options.

It's a `Nitpick` though. 

I was earlier thinking, whether the `Image` class shouldn't be static, so it would rather be `(new Image($file, $options))->url()` then or so. That would avoid the repeated `App::instance()` calls. But also just a thought.